### PR TITLE
Auth/PM-9603 - AnonLayoutWrapper Dynamic content support

### PR DESCRIPTION
**Pull request by JaredSnider-Bitwarden**
## 🎟️ Tracking

[PM-9603](https://bitwarden.atlassian.net/browse/PM-9603)

## 📔 Objective

To add dynamic content support to the `AnonLayoutWrapperComponent`

## 📸 Screenshots

See [storybook](https://62a88a6de5b807fa98886113-smxvyqmtbn.chromatic.com/). 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-9603]: https://bitwarden.atlassian.net/browse/PM-9603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/auth/src/angular/anon-layout/default-anon-layout-wrapper-data.service.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Observable, Subject } from "rxjs";
+
+import { AnonLayoutWrapperDataService } from "./anon-layout-wrapper-data.service";
+import { AnonLayoutWrapperData } from "./anon-layout-wrapper.component";
+
+export class DefaultAnonLayoutWrapperDataService implements AnonLayoutWrapperDataService {
+  private anonLayoutWrapperDataSubject = new Subject<AnonLayoutWrapperData>();
+
+  constructor() {}
```
Comment (jlf0dev) [inline libs/auth/src/angular/anon-layout/default-anon-layout-wrapper-data.service.ts:9]: can be removed

File: libs/auth/src/angular/anon-layout/anon-layout-wrapper-data.service.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Observable } from "rxjs";
+
+import { AnonLayoutWrapperData } from "./anon-layout-wrapper.component";
+
+export abstract class AnonLayoutWrapperDataService {
```
Comment (jlf0dev) [inline libs/auth/src/angular/anon-layout/anon-layout-wrapper-data.service.ts:5]: nit: You have great docs in this but I'd love to see a summary on the class, `AnonLayoutWrapperDataService` is kind of a scary name when you first see it 😁

**Review by jlf0dev — DISMISSED** [review]
nothing blocking, just some suggestions. Great work!

File: libs/auth/src/angular/anon-layout/default-anon-layout-wrapper-data.service.ts (hunk 1 above)
Comment (JaredSnider-Bitwarden) [reply libs/auth/src/angular/anon-layout/default-anon-layout-wrapper-data.service.ts:9]: Done! It was a holdover from the initial stub where I thought I was going to need more deps!

File: libs/auth/src/angular/anon-layout/anon-layout-wrapper-data.service.ts (hunk 2 above)
Comment (JaredSnider-Bitwarden) [reply libs/auth/src/angular/anon-layout/anon-layout-wrapper-data.service.ts:5]: Added a quick summary; let me know if you would add anything else!

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**